src/core/tokenStore.js

```diff
@@ -60,35 +64,30 @@ export default function tokenStoreFactory(options) {
          *
          * @returns {Promise<Object>} the token object
          */
-        dequeue: function dequeue() {
-            var self = this;
-            return self.getIndex().then(function(latestIndex) {
-                var key = _.first(latestIndex);
-                if (!key) return Promise.resolve();
+        dequeue() {
+            return this.getIndex()
+                .then(latestIndex => {
+                    const key = _.first(latestIndex);
```

Comment:
can be latestIndex[0]

Reply:
well, yes and no. This is also a situation where `lodash` is still useful: it will return the first "valid" element.